mods/ra/maps/soviet-10/soviet10.lua

```diff
@@ -0,0 +1,269 @@
+		a.ReturnToBase()
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Reinforcements.Reinforce(USSR, ConvoyEscort, { ConvoyEntry.Location, TanksStop.Location })
+		Media.PlaySpeechNotification(USSR, "ReinforcementsArrived")
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local patrol1 = Reinforcements.Reinforce(Greece, PatrolA, { SouthEntry.Location, PatrolA1.Location })
+		Utils.Do(patrol1, function(unit)
+			Trigger.OnIdle(unit, function(patrols)
+				patrols.Patrol(PatrolAPath, true, 80)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local patrol2 = Reinforcements.Reinforce(Greece, PatrolB, { NorthEntry.Location, PatrolB1.Location })
+		Utils.Do(patrol2, function(unit)
+			Trigger.OnIdle(unit, function(patrols)
+				patrols.Patrol(PatrolBPath, true, 80)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(8), function()
+		Trucks = Reinforcements.Reinforce(Turkey, ConvoyTrucks, { ConvoyEntry.Location, TrucksStop.Location })
+		TrucksFollow()
+	end)
+
+	Trigger.OnEnteredFootprint(LightVehicleAttackTrigger, function(unit, id)
+		if not AttackTriggered and unit.Type == "3tnk" then
+			Trigger.RemoveFootprintTrigger(id)
+			AttackTriggered = true
+			
+			Utils.Do(LightVehicleAttack, function(unit)
+				if not unit.IsDead then
+					IdleHunt(unit)
+				end
+			end)
+		end
+	end)
+end
+
+TrucksFollow = function()
+    Utils.Do(Trucks, function(truck)
+		if not truck.IsDead then
+			-- Trucks that are next to a guard don't need to move
+	        local guards = Map.ActorsInCircle(truck.CenterPosition, WDist.New(2048), IsValidConvoyGuard)
+	        if #guards == 0 then
+	            -- Search nearby area to find a new guard
+	            guards = Map.ActorsInCircle(truck.CenterPosition, WDist.New(6144), IsValidConvoyGuard)
+
+	            if #guards == 0 then
+	                -- Search the whole map as a last resort: this is slow!
+	                guards = Utils.Where(Map.ActorsInWorld, IsValidConvoyGuard)
+	            end
+
+	            if #guards == 0 then
+					RunForIt()
+					return
+				end
+
+	            truck.Move(ClosestActorTo(guards, truck).Location, 3)
+	        end
+	    end
+	end)
+
+    Trigger.AfterDelay(DateTime.Seconds(2), TrucksFollow)
+end
+
+IsValidConvoyGuard = function(actor)
+    -- Ground units (not aircraft) owned by USSR
+    return actor.Owner == USSR and actor.HasProperty("Move") and not actor.HasProperty("Land")
+end
+
+ClosestActorTo = function(actors, target)
+    local closestDistSq = 0
+    local closestActor = nil
+    Utils.Do(actors, function(actor)
+        local offset = actor.Location - target.Location
+        distSq = offset.X * offset.X + offset.Y * offset.Y
+        if closestActor == nil or distSq < closestDistSq then
+            closestDistSq = distSq
+            closestActor = actor
+        end
+    end)
+
+    return closestActor
+end
+
+RunForIt = function()
+	Media.PlaySoundNotification(USSR, "AlertBleep")
```

Comment:
ATM we play the notification and show the message repeatedly once this triggers.
We should instead have a flag to stop `TrucksFollow` from being requeued, and then probably also use an OnIdle trigger to make sure the trucks keep trying to move to the exit if they get nudged or otherwise interrupted.